services/QuillLMS/app/controllers/teacher_infos_controller.rb

```diff
@@ -4,26 +4,7 @@ class TeacherInfosController < ApplicationController
 
   before_action :set_teacher_info, only: [:update]
 
-  def create
-    role = User::TEACHER_INFO_ROLES.include?(session[:role]) ? session[:role] : ''
-    @teacher_info = TeacherInfo.create!(
-      user: current_user,
-      minimum_grade_level: minimum_grade_level,
-      maximum_grade_level: maximum_grade_level,
-      role_selected_at_signup: role,
-      notification_email_frequency: notification_email_frequency
-    )
-
-    subject_areas = SubjectArea.where(id: subject_area_ids)
-    @teacher_info.subject_areas.push(subject_areas)
-
-    render json: {}, status: 200
-  rescue ActiveRecord::RecordInvalid => e
-    render json: {errors: e}, status: 400
-  end
-
```

Comment:
maybe I'm missing something, but I'm not sure if the `role_selected_at_signup` portion of this is accounted for in `create_teacher_info`?

Reply:
@eadams17 you're right, but I deleted this controller action because it wasn't actually getting used anywhere. Some quick investigation suggests that `role_selected_at_signup` is never being set, but I don't know when that became the case. Maybe worth filing a bug report about?